src/pageEditor/tabs/RecipeOptions.tsx

```
@@ -0,0 +1,162 @@
+import { useDispatch, useSelector } from "react-redux";
+import { selectActiveRecipeId } from "@/pageEditor/slices/editorSelectors";
+import { PAGE_EDITOR_DEFAULT_BRICK_API_VERSION } from "@/pageEditor/extensionPoints/base";
+import { useGetRecipesQuery } from "@/services/api";
+import { Formik } from "formik";
+import { useDebouncedCallback } from "use-debounce";
+import { OptionsDefinition } from "@/types/definitions";
+import { actions } from "@/pageEditor/slices/editorSlice";
+
+const fieldTypes = FIELD_TYPE_OPTIONS.filter(
+  (type) => !["File", "Image crop"].includes(type.label)
+);
+
+const formRuntimeContext: RuntimeContext = {
+  apiVersion: PAGE_EDITOR_DEFAULT_BRICK_API_VERSION,
+  allowExpressions: false,
+};
+
+const RecipeOptions: React.VoidFunctionComponent = () => {
+  const [activeField, setActiveField] = useState<string>();
+  const recipeId = useSelector(selectActiveRecipeId);
+  const { data: recipes } = useGetRecipesQuery();
+  const recipe = recipes?.find((recipe) => recipe.metadata.id === recipeId);
+  const initialValues = { optionsDefinition: recipe.options };
+
+  const dispatch = useDispatch();
+  const prevOptions = useRef(initialValues.optionsDefinition);
+  const updateRedux = useDebouncedCallback(
+    (options: OptionsDefinition) => {
+      if (!isEqual(prevOptions.current, options)) {
+        dispatch(actions.editRecipeOptions(options));
+        prevOptions.current = options;
+      }
+    },
+    300,
+    { trailing: true, leading: false }
+  );
+
+  if (recipe.options == null) {
+    return (
+      <Container>
+        <Row>
+          <Col>
+            <Loader />
+          </Col>
+        </Row>
+      </Container>
+    );
+  }
+
+  const noOptions = isEmpty(recipe.options?.schema?.properties);
+
+  return (
+    <div className={styles.paneContent}>
+      <ErrorBoundary>
+        <Formik
+          initialValues={initialValues}
+          onSubmit={() => {
+            console.error(
```

This should throw an error instead of just logging one

---

This is copied exactly from `EditorPane`. Should I change it to `throw` there as well?

---

Yes, I think it makes sense to. Otherwise we wouldn't get error telemetry from it